CodeForPhilly/third-places
# Add location endpoints

# Overview

Adds endpoints for the `location` object with list and detail views. Both of these are GET only since we users won't be adding their own locations. We can add them via the process developed during #22 or via the Django admin.

### Note

I couldn't get tags to work as part of the form and couldn't get hours to behave correctly. I think we will need to revisit our models and serializers to ensure they are what we need them to be

Closes #11 

# Demo

List view:
![image](https://user-images.githubusercontent.com/64238570/236856471-c9231136-4b57-4238-bd81-470eb65688b9.png)

Detail view: 
![image](https://user-images.githubusercontent.com/64238570/236856638-f14d7c2c-8492-4f58-9588-ccb562e086bf.png)

# Testing instructions

- Create a super user
  - In one terminal tab run `docker-compose up --build` and leave that open
  - In a new terminal tab:
    - run 'docker exec -it third-places-django-1 bash`
    - Once in the django container's shell, run `python3 manage.py createsuperuser`
    - Follow the form to create a superuser with any name, password, etc. This is just for you
    - Exit the container shell with CTRL-D
- In your browser, go to [http://localhost:8321/admin/](http://localhost:8321/admin/)
  - Add 1 or 2 locations with the "Add" button for locations
- Navigate to [http://localhost:8321/api/locations](http://localhost:8321/api/locations/) and ensure:
  - [x] You see the locations you made
- Navigate to [http://localhost:8321/api/locations/1](http://localhost:8321/api/locations/1) and ensure:
  - [x] You see the location you made
- Navigate to [http://localhost:8321/api/locations/200](http://localhost:8321/api/locations/200) and ensure:
  - [x] You see a 404 response

# Checklist before merge
